src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/LagPortSpec.groovy

```diff
@@ -516,6 +516,70 @@ class LagPortSpec extends HealthCheckSpecification {
         lagPort && northboundV2.deleteLagLogicalPort(sw.dpId, lagPort)
     }
 
+    @Tidy
+    def "Able to create/update LAG port with duplicated port numbers on the #sw.hwSwString switch"() {
+        when: "Create LAG port with duplicated port numbers"
+        def switchPortToCreate = topology.getAllowedPortsForSwitch(sw)[0]
+        def swAddress = northbound.getSwitch(sw.dpId).address
+        def portListToCreate = [switchPortToCreate, switchPortToCreate]
+        def createPayload = new LagPortRequest(portNumbers: portListToCreate)
+        def lagPortCreateResponse = northboundV2.createLagLogicalPort(sw.dpId, createPayload)
+
+        then: "Response shows that LAG port created successfully"
+        with(lagPortCreateResponse) {
+            logicalPortNumber > 0
+            portNumbers == [switchPortToCreate]
+        }
+        def lagPort = lagPortCreateResponse.logicalPortNumber
+
+        and: "Northbound shows that LAG port created"
+        with(northboundV2.getLagLogicalPort(sw.dpId)[0]) {
+            logicalPortNumber == lagPort
+            portNumbers == [switchPortToCreate]
+        }
+
+        and: "GRPC shows that created port exist in a list of all LAG ports"
+        with(grpc.getSwitchLogicalPortConfig(swAddress, lagPort)) {
+            logicalPortNumber == lagPort
+            name == "novi_lport" + lagPort.toString()
+            portNumbers == [switchPortToCreate]
+            type == LogicalPortType.LAG
+        }
+
+        when: "Update the LAG port with duplicated port numbers"
+        def switchPortToUpdate = topology.getAllowedPortsForSwitch(sw)[1]
```

Comment:
`topology.getAllowedPortsForSwitch(sw)` you can extract it in a separate variable, it is already used twice.

Reply:
also, there is some chance that `topology.getAllowedPortsForSwitch(sw)` return just one element. It can be handled early.
I would check the size of the array at the beginning of the test  and remove L551 `assert switchPortToCreate != switchPortToUpdate`